### ISSUE TITLE
Completion Enum insertion

### ISSUE THREAD

**Issue by mkrueger** [dotnet/roslyn#19506]
```csharp
enum FooBar {  Foo, Bar }
class Program
{
     static void Main(string[] args)
     {
          FooBar foo = 
     }
}
```

Would be nice if FooBar.Foo and FooBar.Bar would be in the completion list. Makes things easier to type.

**Comment by sharwell** [comment]
@CyrusNajmabadi @Pilchie Is this something was considered and rejected in the past, or is it new? At first glance, seems like a good idea to me.

**Comment by Pilchie** [comment]
That's basically what VB does.  C# just preselects `FooBar` so that typing `.` is enough to get you to the members.  There is not real reason that C# doesn't flatten the list except that it historically never has.

**Comment by CyrusNajmabadi** [comment]
It would be fine with me.  Ideally VB and C# would be able to emulate each other here, and there would be an option to specify the type of behavior you want.  My muscle memory is such that i just expect to type "space-dot-membername", so as long as you don't break that, i'm fine :)

**Comment by mkrueger** [comment]
Doesn't make sense to break space-dot-membername. If the items are in the list that doesn't interfere with that. But once you're used to the behavior above it's hard going back :)

**Comment by alrz** [comment]
> i just expect to type "space-dot-membername", 

That's too much typing compared to just clicking arrows and selecting the member.

I see this as an alternative to a language feature that allows `FooBar x = Foo;`.

**Comment by sharwell** [comment]
This was implemented under issue #47193
